src/components/UsersSuggestionDropdown.js

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useRef, useState } from 'react'
+import axios from 'axios'
+import { trackPromise, usePromiseTracker } from 'react-promise-tracker'
+import { apiURL } from '../config.json'
+import LoadingIndicator from '../modules/LoadingIndicator'
+
+const UsersSuggestionDropdown = ({
+  isOpen,
+  userTerm,
+  setUserTerm,
+  setUserQuery,
+  setUsersSuggestionOpen
+}) => {
+  const [usersToSuggest, setUsersToSuggest] = useState([])
+
+  const { promiseInProgress } = usePromiseTracker({
+    area: 'user-suggest-dropdown'
+  })
+
+  const dropDown = useRef(null)
+
+  const cancelToken = useRef()
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (dropDown.current && !dropDown.current.contains(event.target)) {
+        setUsersSuggestionOpen(false)
+      }
+    }
+    document.addEventListener('mousedown', handleClickOutside)
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside)
+    }
+  }, [setUsersSuggestionOpen])
+
+  useEffect(() => {
+    if (userTerm.length <= 0) {
+      setUsersSuggestionOpen(false)
+      return
+    }
+
+    if (typeof cancelToken.current !== typeof undefined) {
+      cancelToken.current.cancel()
+    }
+
+    cancelToken.current = axios.CancelToken.source()
+
+    const fetchUsers = async () => {
+      const response = await axios.post(
+        `${apiURL}/graphql`,
+        {
+          query: `query {
+            users (where: {
+              username_contains: "${userTerm}"
+            }) {
+              id
+              username
+              profilePicture {
+                url
+              }
+            }
+          }`
+        },
+        { cancelToken: cancelToken.current.token }
+      )
+      setUsersToSuggest(response.data.data.users)
+    }
+    if (userTerm.length > 0) {
+      trackPromise(fetchUsers(), 'user-suggest-dropdown')
+    } else {
+      setUsersToSuggest([])
+    }
+  }, [userTerm, setUsersSuggestionOpen])
```

Comment:
do we really need this? these are props and components will re-render anyway on props change

Reply:
I think yes, because of `setUsersToSuggest(response.data.data.users)` if we remove the dependencies then, it will be called again & again.